### PR TITLE
[Feature] Add an e2e test for K8s Job submitter failures

### LINKED ISSUE #2655: [Feature] Add an e2e test for K8s Job submitter failures
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

Add an test for https://github.com/ray-project/kuberay/pull/2579.

1. Create a RayJob
2. Wait for the Pod created by the K8s Job submitter to be ready
3. Let the Pod to run for a while (e.g. 15s)
4. Delete the Pod
5. The K8s Job should create another Pod
6. The RayJob should eventually succeed.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by simotw**
## Why are these changes needed?
Add an test for https://github.com/ray-project/kuberay/pull/2579.

1. Create a RayJob
2. Wait for the Pod created by the K8s Job submitter to be ready
3. Let the Pod to run for a while (e.g. 15s)
4. Delete the Pod
5. The K8s Job should create another Pod
6. The RayJob should eventually succeed.
7. The RayJobDeployment eventually complete. 
(Add this according to the issue https://github.com/ray-project/kuberay/issues/2154)

CleanShot 2024-12-26 at 15 15 00@2x

## Related issue number

Close #2655
## Checks

- [X] I've made sure the tests are passing.
- Testing Strategy
   - [X] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
cc @rueian would you mind reviewing this PR? Thanks!

File: ray-operator/test/e2e/rayjob_recovery_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				PropagationPolicy: &propagationPolicy,
+			})
+			g.Expect(err).NotTo(HaveOccurred())
+		}
+
+		// Get old pod names for comparison
+		oldPodNames := make([]string, len(jobpods.Items))
+		for i, pod := range jobpods.Items {
+			oldPodNames[i] = pod.Name
+		}
```
Comment (rueian) [inline ray-operator/test/e2e/rayjob_recovery_test.go:74]: It seems that `oldPodNames` is not used below.

Comment (simotw) [reply ray-operator/test/e2e/rayjob_recovery_test.go:74]: Oh, I forgot I chose to use `jobpods.Items` directly. Thank you!

**Review by rueian — APPROVED** [review]

**Comment by rueian** [comment]
LGTM.

I just also come up with a new idea of not relying on the sleep operation but using the named actor for this e2e test: the job script can exit with an manual error after registering a named actor to Ray and later being restart by k8s it can exit normally as the named actor is already registered.

This way we don’t need to wait for 15 seconds to delete the pod and 1 more minute for job completion.

What do you think @simotw, @kevin85421? Maybe this can be another PR.

**Comment by rueian** [comment]
Sorry, please ignore the above idea. It doesn’t make submitter fail.

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**